enhancements/sig-policy/130-policy-merge-compliancetypes/README.md

```diff
@@ -0,0 +1,376 @@
+# ConfigurationPolicy ComplianceTypes for different merge behaviors
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+We propose to add new options for the `spec["object-templates][*].complianceType` field in
+ConfigurationPolicies to allow the user to specify new merge strategies: `musthaveapply`, 
+`musthavestrategic` and `musthavemerge`. These new options correspond to existing options in
+`kubectl`, which users may be familiar with. Along with the existing `musthave`, `mustonlyhave`, and
+`mustnothave` options, this should provide more flexibility and control to users, and be able to
+provide a more predictable experience.
+
+## Motivation
+
+The existing `musthave` complianceType has a bespoke algorithm for merging content from the cluster
+state with what is defined in the policy. One specific difference is when the policy includes a list
+in the object template: `musthave` will check the existing list for any items specified in the
+policy, and if any are missing, it will *add* them to the list and apply the change. Tools like
+`kubectl apply` can do something similar on some objects, via a "strategic merge", but not on custom
```

Comment:
I think strategic merge does work on custom resources:
https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch

Reply:
I take that back. I got confused with this:
https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy

Reply:
Oh, good to know that server-side-apply can do strategic merges on custom resources, although it begs the question of why exactly the client-side can't do the same thing - it should have access to the same openapi information from the CRDs...

I'll have to try server-side-apply more, but from a quick read I'd be worried it would have more "conflicts" than we'd want?